modules/diagnostic/src/VerifyMapping.php

```diff
@@ -0,0 +1,183 @@
+   *
+   * @return array
+   *   Information about the mapping and its verification.
+   */
+  public function verify($acmUuid = "") {
+
+    $this->debugLogger("Did call 'verify' function");
+    $systemAdvice = "";
+    $passedSoFar = TRUE;
+
+    if (!$acmUuid) {
+      $passedSoFar = FALSE;
+      $systemAdvice .= "\nMissing ACM UUID from header in request to Drupal.";
+    }
+
+    $response = [];
+    $response['acm_uuid'] = $acmUuid;
+
+    $response['locale'] = $this->i18nHelper->getLangcodeFromStoreId($acmUuid);
+
+    $locale = $this->languageManager->getCurrentLanguage()->getId();
+    $systemAdvice .= "The current language of this endpoint is: ";
+    $systemAdvice .= $locale . ".";
+    $acm_uuid = $this->i18nHelper->getStoreIdFromLangcode();
+    if (!$acmUuid) {
+      $passedSoFar = FALSE;
+      $systemAdvice .= "\nACM UUID not associated with this locale.";
+    }
+
+    if ($acm_uuid !== $response['acm_uuid']) {
+      $passedSoFar = FALSE;
+      $systemAdvice .= "\nThe ACM_UUID associated with the language of the";
+      $systemAdvice .= " Drupal URL, '" . $acm_uuid . "', does not match the";
+      $systemAdvice .= " ACM_UUID of the header";
+      $systemAdvice .= " sent from Commerce Middleware, ";
+      $systemAdvice .= $response['acm_uuid'] . ".";
+    }
+
+    if ($locale !== $response['locale']) {
+      $passedSoFar = FALSE;
+      $systemAdvice .= "\nThe language of the";
+      $systemAdvice .= " Drupal URL, '" . $locale . "', does not match the";
+      $systemAdvice .= " language defined by the ACM_UUID";
+      $systemAdvice .= " sent from Commerce Middleware, ";
+      $systemAdvice .= $response['acm_uuid'] . ".";
+    }
+
+    // This URL.
+    // urlGenerator is a service. consider:
+    // getContainer()->get('url_generator')
+    $response['system_api_url'] = \Drupal::urlGenerator()->generateFromRoute('<front>', [], ['absolute' => TRUE]);
+
+    // Connector URL.
+    $configAcmConnector = $this->configFactory->get('acm.connector');
+    $response['connector_api_url'] = $configAcmConnector->get('url');
+    if ($response['connector_api_url'] == "") {
+      $passedSoFar = FALSE;
+      $systemAdvice .= "\nMissing connector URL in Drupal config.";
+    }
+
+    // Currency information.
+    $configAcmCurrency = $this->configFactory->get('acm.currency');
+    $currency = \Drupal::service('repository.currency')->get($configAcmCurrency->get('currency_code'), $locale, 'en');
+    $response['default_currency'] = $currency->getCurrencyCode() . " (" . $currency->getName() . ")";
+
+    // Description (site description in locale?)
+    $configSystemSite = $this->configFactory->get('system.site');
+
+    // Translations? Is the config locale aware?
+    $description = $configSystemSite->get('name');
+    $description .= " (" . $configSystemSite->get('slogan') . ").";
+    $response['description'] = $description;
+
+    // TODO (malachy): Add in the loop back to test connection
```

Comment:
Same question as in commerce-manager-magento counterpart: is this something you want to implement in this PR? Just want to make sure it doesn't miss any functionality. :)

Reply:
There will be a follow-on story to implement it.

Reply:
Yeah, heard it on the demo, sorry.